File: app/src/main/java/io/catter2/di/SharedPrefFavoritesRepoDIModule.java
package io.catter2.di;

import android.content.Context;

import io.catter2.favorites.FavoritesRepository;
import io.catter2.favorites.SharedPrefFavoritesRepository;

public class SharedPrefFavoritesRepoDIModule extends FavoritesRepoDIModule {
    private String userToken;

    public SharedPrefFavoritesRepoDIModule(String userToken) {
        this.userToken = userToken;
    }

    @Override
    public FavoritesRepository provideFavoritesRepository(Context appContext, String userToken) {
        return new SharedPrefFavoritesRepository(appContext, userToken);
    }

    @Override
    public String provideUserToken() {
        return this.userToken;
    }
}

File: app/src/main/java/io/catter2/di/SharedPrefFavoritesRepoDIModule.kt
package io.catter2.di

import android.content.Context

import io.catter2.favorites.FavoritesRepository
import io.catter2.favorites.SharedPrefFavoritesRepository

class SharedPrefFavoritesRepoDIModule(private val userToken: String) : FavoritesRepoDIModule() {
    override fun provideFavoritesRepository(appContext: Context, userToken: String): FavoritesRepository =
            SharedPrefFavoritesRepository(appContext, userToken)

    override fun provideUserToken(): String = this.userToken
}
